Volume slicing

AnniekStok opened the pull request:
Work in progress for the new module on volume slicing. Happy to discuss which improvements are necessary here. Right now the orthogonal slices is in the activity and reslicing in the exercise, but maybe it should go together.

tischi commented on _includes/volume_slicing/exercises/volume_slicing_gui.md:
```diff
@@ -0,0 +1,14 @@
+Open the multidimensional image [xyz_16bit_t1-head.tif](https://github.com/NEUBIAS/training-resources/raw/master/image_data/xyz_16bit_t1-head.tif).
+- Reslice the image from different directions, such that you can view the image stack from -the side, from the top, and from the front.
```
Yes, typically exercises should be easy and just repetitions of the activity with different data. Thus, adding Reslice to the activity would make sense. It would also be great to find a biological relevant data set for this.
@ssgpers do you have some data where (maybe from Dima's project) where it makes sense to reslice in a non-orthogonal direction?

AnniekStok replied:
Ok, will change this! I guess something like this will do? 
1. Visualize data using orthogonal views function
2. Subset part of the stack
3. Orthogonal reslicing
4. Non-orthogonal reslicing

tischi commented on _includes/volume_slicing/exercises/volume_slicing_gui.md:
```diff
@@ -0,0 +1,14 @@
+> - Select 'Top' in the 'Start at:' drop-down menu to reslice the image from the top and view the head from above.
+> - Select 'Left' in the 'Start at:' drop-down menu to reslice the image from the left and view the head from the front.
+> - Tick the 'Rotate 90 degrees' button to rotate the output image (you can use this when reslicing from the left, so that the top of the head remains pointing upwards.)
+> - Ticking the 'Avoid interpolation box' matters when you have anisotropic data. In this case the data is isotropic (voxels are 1.5 x 1.5 x 1.5 mm), so it has no effect, unless you specified a different output spacing in the 'Output spacing' box at the top of the menu. You can view what happens when you change the voxel dimensions in the properties window (**[Ctrl-Shift-P]**). For example, set t ...
```
I feel that this is too complicated. Should be taught first in the activity.

tischi commented on _modules/volume_slicing.md:
```diff
@@ -0,0 +1,59 @@
+
+concept_map: >
+  graph TD
+    DM("Image data dimensions") --> S("Spatial dimensions")
```
I think the slicing concept is missing a bit. Something like:


N-Dim image --> slice --> N-1 Dim image

Not sure...

tischi commented on _modules/volume_slicing.md:
```diff
@@ -0,0 +1,59 @@
+
+    > ## Solution
+    > - Isotropic image data has voxels of equal XYZ dimensions. **True**
+    > - Reslicing volumetric data is essentially similar to rotating a 3D object and viewing it from a different angle. **True**
```
I would not agree.... One really subsets the data and not only looks at it from a different angle.

AnniekStok replied:
Ok I will change this. I wrote it like this because I was using the 'Reslice' function which does not subset the stack, but indeed there are other occasions where one can do one or both (rotating and/or subsetting).

tischi reviewed (commented):
Super nice! 

I think the "slicing" concept should be more apparent.

Also in the figure.

@stemarcotti do you maybe have something to demonstrate how to, e.g., extract a 2D slice from a 3D image.

AnniekStok commented:
@tischi Thanks for your comments! I will work on it more later this week. Indeed the slicing could be expanded. I will try to change it such that it first explains possible dimensions, and then basic processes including both the Reslice function and subsetting a part of the stack.

tischi commented:
@AnniekStok @manerotoni @stemarcotti 

I thought about this module and maybe the concept that we want to teach here is:

`N-Dim Image Data --- show on ---> 2D Monitor`

That means, technically that we always have to subset the, e.g. 5D, data such that it can be represented on a 2D monitor.

For a 3D image the explanatory figure could look something like this (obviously there must be a more beautiful way to show this):

![image](https://user-images.githubusercontent.com/2157566/149794494-298d018f-0552-45bb-846e-cb3be14ad07b.png)

There is also the concept `N Dim  Data --- slice ---> N-1 Dim Data` as for example implemented in ImgLib2 `Views.hyperslice()`, but my current thinking is that for this module, which could only be about visualization, we could restrict ourselves to the special case that we want 2D data in the end.

What do you think?

stemarcotti commented:
> Super nice!
> 
> I think the "slicing" concept should be more apparent.
> 
> Also in the figure.
> 
> @stemarcotti do you maybe have something to demonstrate how to, e.g., extract a 2D slice from a 3D image.

@tischi I have nothing ready but could put it together! I will start with your draft idea and then we can modify it in case, shouldn't take me long, if that's alright?

tischi commented:
@stemarcotti Would be awesome! Thanks a lot! Ideal would be PowerPoint...as not all of us have illustrator.

stemarcotti commented:
<img width="783" alt="Screenshot 2022-01-19 at 10 27 20" src="https://user-images.githubusercontent.com/33655510/150112440-e2b4e107-d5c3-44ce-a874-11c07aaac35b.png">
[20220118_slicing.pptx](https://github.com/NEUBIAS/training-resources/files/7896225/20220118_slicing.pptx)


@AnniekStok @tischi let me know if something like this could work :)

tischi commented:
@stemarcotti 

I ❤️ it!  🥳 

Especially adding the notion that one has to turn the image to show it on the 2D screen.

One thing: the perspective here:

<img width="260" alt="image" src="https://user-images.githubusercontent.com/2157566/150122512-8b704163-fb3b-40e0-9ed8-9cf60dad0c46.png">

Can you shear it such that it looks the same as the other one?

<img width="312" alt="image" src="https://user-images.githubusercontent.com/2157566/150122600-ff54c505-2419-4a3f-80e3-33ef4ffd301e.png">

stemarcotti commented:
<img width="722" alt="Screenshot 2022-01-19 at 11 48 38" src="https://user-images.githubusercontent.com/33655510/150124101-da818b60-d944-433d-9414-b1f355c7f076.png">
[20220118_slicing.pptx](https://github.com/NEUBIAS/training-resources/files/7896791/20220118_slicing.pptx)


well spotted, thanks @tischi!

manerotoni commented:
Nice!

AnniekStok commented:
Beautiful!! Thanks a lot!!

AnniekStok commented:
I have updated the module: 

1) updated the motivation
2) Updated the concept map following @tischi 's suggestion.
3) Updated the figure. Thank you @stemarcotti  for your contribution! Somehow I could not find the pptx file in the path you mentioned ((https://github.com/NEUBIAS/training-resources/files/7896225/20220118_slicing.pptx), the folder 'files' is not visible to me. Therefore I just pasted it in my pptx file as a PNG, but I think it would be nice to have the editable one for the future. 
4) Updated the activity using the beads image: 
- inspect image using the sliders
- inspect image using orthogonal views
- subset the image by cropping out the green bead
- Reslice the green bead from the top to see it from the other side, and compare the result when you tick or untick the 'avoid interpolation box'
5) Updated the exercise: viewing and reslicing the 'head' stack from different angles, followed by an example of non-orthogonal reslicing to see the head in 3/4 view (happy to change this if needed. I thought alternatively I could make an image of a simple object (e.g. a &  sign or something), but rotated in weird angle in 3D so that it is not immediately obvious what it is. The exercise would be to rotate and reslice it until you can see the sign from the front so that it is readable. )

I realized there is some overlap with the spatial calibration module, I think @tischi already made an issue about this. Should I change anything about this in the volume_slicing module?

I still need to think of another true/false question...

I realized the word 'slice' is used in different contexts which is potentially confusing. I tried to explain it below the figure.
- the word 'slice'  as in z-slice to indicate the location in the z-dimension
- slicing as in subsetting part of a multidimensional stack
- the term 'reslicing' is used in Fiji to indicate resampling of the image from a different angle (but it does not subset). 

Hope this helps, I am happy to make further updates where necessary.

tischi commented:
@manerotoni Would you have time to review and merge this? I am in meetings today and tomorrow...

manerotoni commented:
Not before tomorrow evening.

stemarcotti commented:
@tischi and @manerotoni let me know if you can see the ppt somewhere when you review this, otherwise happy to share it in a different way! Have a good week ahead everyone =)

tischi commented:
Thanks a lot @AnniekStok.
I merge now.

AnniekStok commented:
Thank you @tischi !

On the website, somehow the figure is missing, I don't know why. I can see it properly with my local jekyll generated version that I pushed yesterday. Any ideas?

AnniekStok commented:
Ah, I think it is because it is somehow accidentally a PNG and not a png... I will change the extension and try to push again...

tischi commented:
@AnniekStok 
You'll have to rename in a special way:
https://stackoverflow.com/questions/17683458/how-do-i-commit-case-sensitive-only-filename-changes-in-git
